packages/react-native-tracer-provider/android/src/main/java/io/embrace/tracerprovider/TracerProviderModule.kt

```diff
@@ -0,0 +1,288 @@
+            val link = links.getMap(i)
+            val linkSpanContext = link.getMap(LINK_SPAN_CONTEXT_KEY) ?: continue
+            val linkAttributes = link.getMap(LINK_ATTRIBUTES_KEY)
+
+            if (linkAttributes == null) {
+                val spanContext = spanContextFromReadableMap(linkSpanContext)
+                spanBuilder.addLink(spanContext)
+            } else {
+                spanBuilder.addLink(spanContextFromReadableMap(linkSpanContext), attributesFromReadableMap(linkAttributes))
+            }
+        }
+
+        // Set parent
+        val parent = spans[parentID];
+        if (parentID.isEmpty() || parent == null) {
+            spanBuilder.setNoParent()
+        } else {
+            spanBuilder.setParent(parent.storeInContext(Context.root()))
+        }
+
+        val span = spanBuilder.startSpan();
+        spans[spanBridgeID] = span;
+        promise.resolve(spanContextToWritableMap(span.spanContext))
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true) fun spanContext(spanBridgeID: String) : WritableMap {
+        return spanContextToWritableMap(spans[spanBridgeID]?.spanContext)
+    }
+
+    @ReactMethod fun setAttributes(spanBridgeID: String, attributes: ReadableMap) {
+        val span = spans[spanBridgeID] ?: return;
+        span.setAllAttributes(attributesFromReadableMap(attributes));
+    }
+
+    @ReactMethod fun addEvent(spanBridgeID: String, eventName: String, attributes: ReadableMap, time: Double) {
+        val span = spans[spanBridgeID] ?: return;
+
+        if (time != 0.0) {
+            span.addEvent(eventName, attributesFromReadableMap(attributes), time.toLong(), TimeUnit.MILLISECONDS)
+        } else {
+            span.addEvent(eventName, attributesFromReadableMap(attributes))
+        }
+    }
+
+    @ReactMethod fun addLinks(spanBridgeID: String, links: ReadableArray) {
+        val span = spans[spanBridgeID] ?: return;
+        for(i in 0..<links.size()) {
+            val link = links.getMap(i)
+            val linkSpanContext = link.getMap(LINK_SPAN_CONTEXT_KEY) ?: continue
+            val linkAttributes = link.getMap(LINK_ATTRIBUTES_KEY)
+
+            if (linkAttributes == null) {
+                span.addLink(spanContextFromReadableMap(linkSpanContext))
+            } else {
+                span.addLink(spanContextFromReadableMap(linkSpanContext), attributesFromReadableMap(linkAttributes))
+            }
+        }
+    }
+
+    @ReactMethod fun setStatus(spanBridgeID: String, status: ReadableMap) {
+        val span = spans[spanBridgeID] ?: return;
+        val statusCode = status.getString(SPAN_STATUS_CODE_KEY) ?: return
+        val message = status.getString(SPAN_STATUS_MESSAGE_KEY) ?: ""
+
+        try {
+            if (message.isEmpty()) {
+                span.setStatus(StatusCode.valueOf(statusCode))
+            } else {
+                span.setStatus(StatusCode.valueOf(statusCode), message)
+            }
+        } catch(e: IllegalArgumentException) {}
```

Comment:
Can we log this rather than just swallowing it?

Reply:
will add